src/android/com/silkimen/http/OkConnectionFactory.java

```diff
@@ -1,26 +1,78 @@
 package com.silkimen.http;
 
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import okhttp3.CertificatePinner;
 import okhttp3.OkHttpClient;
 import okhttp3.OkUrlFactory;
 
-import java.net.URL;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
 import java.net.HttpURLConnection;
-import java.net.URLStreamHandler;
 import java.net.Proxy;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
 
 public class OkConnectionFactory implements HttpRequest.ConnectionFactory {
-  private final OkHttpClient client = new OkHttpClient();
 
-  public HttpURLConnection create(URL url) {
-    OkUrlFactory urlFactory = new OkUrlFactory(this.client);
+  private final OkHttpClient client;
+
+  public OkConnectionFactory() {
+    // Load JSON and initialize the OkHttpClient with certificate pinning
+    try {
+      String jsonConfig = loadJson("assets/certificate_settings.json");
```

Comment:
This code is executed on every request.
We should add the parsing of the json to a section, that is being called once only.

instantiating the client on every request is ok, (was done like this before)